### ISSUE TITLE
[Bug Report]: NFC reads gets the system stuck

### ISSUE THREAD

**Issue by JeDaYoshi** [ndeadly/MissionControl#875]
### Switch Firmware Version

19.0.0 (Latest)

### Atmosphère Version

1.8.0 (Latest)

### Mission Control Version

0.12.0 (Latest)

### Boot Method

Hekate

### Issue Description

After updating to Mission Control 0.12.0, I ran into issues while using my DualShock 4.

As soon as a game attempted to use NFC, whether for amiibo or not, everything would freeze - CPU 4 would get stuck at 100% (according to Status Monitor Overlay), not even letting me go back to HOME, until the game stopped attempting to read NFC.
I ran into this issue with Splatoon 3 (which crashes once you try to switch zones when this issue happens, probably unrelated as it struggles to load data), and Sky: Children of the Light (using the STAR scan function).

To reproduce, you just have to use anything that tries to read NFC.

### Additional Context

After testing, seems like commit 6ce89a7f54928c7a2fa7020bca047608439f470e is the culprit.

I reverted this commit and recompiled. Since then, I no longer experience the issue and Mission Control returns to the old behaviour prior to this update (aka. "An error occurred while trying to access the amiibo.").

**Comment by JeDaYoshi** [comment]
Changed the title as this might not be an issue specific to DS4.

**Comment by ndeadly** [comment]
Hmm, that commit was supposed to fix all these nfc issues. I was only able to test with Splatoon 2 though. Will look into it

**Comment by k1gs** [comment]
I have the same problem too. 

HOS:
* [19.0.0]
Atmo:
* 1.8.0 
Mc version:
* 0.12.0

This happens when the Dualsense controller is connected in Splatoon 3 when entering the wardrobe in the game, and the NFC box. I will separately attach a video of how the [bug](https://youtu.be/ladfCgmWBw0) manifests itself, I hope it will help. I'm sorry that I can get into this Issue like this, I just thought I could report the problem right here

**Comment by tthe3d** [comment]
Can you share the version you compiled without the commit? For the people that run 19.0.1  ..  This is unusable in certain games like splatoon 3.

**Comment by JeDaYoshi** [comment]
Ah... I swore I did, since I had thought of it after noticing there hasn't been any new release yet.

There you go.
[MissionControl-0.12.0-master-4ed9ebc.zip](https://github.com/user-attachments/files/17963440/MissionControl-0.12.0-master-4ed9ebc.zip)

**Comment by tthe3d** [comment]
Thank you !!

**Comment by tthe3d** [comment]
I just found out about a temporal fix for the error message. Turn on airplane mode , enable Internet and bluetooth leaving NFC disabled. This way console doesn't look for NFC signals.

**Comment by CountDarkk** [comment]
Idk if i'm stupid or not (i'm not a tech genius), but have you tried to use something to replace the console's function so that instead of reading the nfc signal from the controller it reads the nfc signal from apps like emuiibo? (idk, just what i thought woukd be, don't judge me).

**Comment by CountDarkk** [comment]
Oh wow. The build..actually fixed it.

**Comment by CountDarkk** [comment]
I've tested all the button and functions.. and it seems it has no bugs. I think you can post it as build 0.12.1.

**Comment by ndeadly** [comment]
Cool, I will integrate this into the 0.13.0  release for 20.0.x support then.

I've known about the potential fix for a while, but licensed controllers have some of the official features blocked (button remapping, stick calibration etc). I recently found a way to patch it to support those features though, so i should be able to add them back (and as a bonus support them for real licensed controllers too)

**Comment by CountDarkk** [comment]
Yeah, but as far as i know i don't think atmosphere or hekate support firmware 20.0.x. Try leaving it for 19.0.x for now.

**Comment by ndeadly** [comment]
They both do, it's just in the testing phase. I won't be making my release until Atmosphere does anyway. And besides, Mission Control builds are backwards compatible. The one I just shared is built with support for 20.0.x already and as you've just seen, works for you too.

**Comment by CountDarkk** [comment]
Also idk if it's a bug but when i am in a game and start moving my character using the joycons the controller automatically disconnects.

**Comment by ndeadly** [comment]
Unlikely a bug. MC just facilitates connecting the controller, and transforming everything to look like it comes from a Nintendo controller. It doesn't manage connections in any way. This is probably just a game enforcing some controller policy.

**Comment by CountDarkk** [comment]
Hi. I was just wondering if you patched the build. Idk if all people know about this open issue, so they might still have issues with their controllers. I suggest that if you've patched the build to post it as version 0.13.0 and close this issue.

**Comment by ndeadly** [comment]
Should be fixed in https://github.com/ndeadly/MissionControl/releases/tag/v0.13.0
